src/main/java/com/squareup/kotlinpoet/FunSpec.kt

```diff
@@ -283,6 +297,26 @@ class FunSpec private constructor(builder: Builder) {
       parameters += parameterSpec
     }
 
+    fun callThisConstructorWithParameters(vararg parameters: String) =
```

Comment:
This name is a bit much. Can we do any better? Maybe `delegateConstructor`?

Reply:
`this` and `super` still have to be in the name, what about `delegateToThisConstructor` and `delegateToSuperConstructor`?

Reply:
We could expose an enum that lets you pick between the two delegates.

Reply:
Can do. I hope to try `ConstructorSpec` soon, part of the problem with naming comes from the need to clarify that the methods should be used with the `constructorBuilder()`.